File: cmd/server/monitor.go
package main

import (
	"github.com/guarandoo/neko/pkg/core"
	"github.com/guarandoo/neko/pkg/notifier"
	"github.com/guarandoo/neko/pkg/probe"
)

type Monitor struct {
	Name      string
	Interval  string
	Probe     probe.Probe
	Status    core.Status
	Notifiers []notifier.Notifier
}

File: src/monitor.rs
use crate::config::MonitorConfig;
use crate::core::Status;
use crate::notifier::Notifier;
use crate::probe::Probe;

pub struct Monitor {
    pub name: String,
    pub interval: String,
    pub probe: Box<dyn Probe + Send + Sync>,
    pub status: Status,
    pub notifiers: Vec<Box<dyn Notifier + Send + Sync>>,
    pub configuration: MonitorConfig,
}
